back/engines/commercial/analysis/spec/services/inputs_finder_spec.rb

```diff
@@ -313,6 +313,32 @@
     end
   end
 
+  describe 'input_custom_field_no_empty_values' do
+    let_it_be(:analysis) { create(:analysis) }
+    let_it_be(:custom_form) { create(:custom_form) }
+    let_it_be(:custom_field_select) { create(:custom_field_select, :with_options, resource: custom_form) }
+
+    let_it_be(:input0) { create(:idea, project: analysis.source_project) }
+    let_it_be(:input1) do
+      create(:idea, project: analysis.source_project, custom_field_values: {
+        custom_field_select.key => custom_field_select.options[0].key,
+      })
+    end
+
+    let_it_be(:input2) do
+      create(:idea, project: analysis.source_project, custom_field_values: {
+        custom_field_select.key => custom_field_select.options[1].key,
+      })
+    end
+
+    it 'filters out custom_field with no empty values correctly' do
```

Comment:
@sebastienhoorens This spec is failing (the input is not getting filtered) but I can't figure out why. As far as I can tell, the inputs are setup correctly, maybe I'm missing something obvious here? Could you help me out 🙏

Reply:
@IvaKop I don't see any custom fields being linked to the analysis, therefore I expect the filter will loop over an empty list and filter out nothing. If you do something like `create(:analysis, custom_fields: [custom_field_select])` on line 317, hopefully it will work.

Reply:
That was indeed the problem, I missed that I had to add the custom fields to the analysis, as well. Thanks! 🙌 The PR is now fully ready for review, I think.